custom/icds_reports/utils/aggregation_helpers/distributed/agg_service_delivery_report.py

```diff
@@ -286,6 +286,42 @@ def staging_queries(self):
             'next_month_start_date': next_month_start
         }
 
+        yield f"""
+        UPDATE "{self.temporary_tablename}" agg_sdr SET
+            breakfast_served = ut.breakfast_served,
+            hcm_served = ut.hcm_served,
+            thr_served = ut.thr_served,
+            pse_provided = ut.pse_provided,
+            breakfast_21_days = ut.breakfast_21_days,
+            hcm_21_days = ut.hcm_21_days,
+            pse_16_days = ut.pse_16_days,
+            breakfast_9_days = ut.breakfast_9_days,
+            hcm_9_days = ut.hcm_9_days,
+            pse_9_days = ut.pse_9_days
+
+        FROM (
+            SELECT
+                awc_id,
+                supervisor_id,
+                sum(open_bfast_count) AS breakfast_served,
+                sum(open_hotcooked_count) AS hcm_served,
+                sum(days_thr_provided_count) as thr_served,
+                sum(open_pse_count) as pse_provided,
+                CASE WHEN (sum(open_bfast_count) >=21) THEN 1 ELSE 0 END AS breakfast_21_days,
```

Comment:
is it possible to do these at query time. the day boundaries are likely to be different if we ever have to do this in another report (in fact they already are different than the ones we have discussed for the additions to the service delivery dashboard). Rather than having to add a new column for any possible amount of days, it would be easier to just check the count when we render the report.

Reply:
This is especially true since you already iterate over the whole result set in the MPR query

Reply:
we could pull it but we would need to query the agg_level 5 everytime. Do you think that should be fine for web report  to pull data from this table.